api/classes/publisher.js

```diff
@@ -1,31 +1,143 @@
-// var AWS = require('aws-sdk');
-// var s3 = new AWS.S3();
+var Promise = require('bluebird');
+
+var noxmox = require('noxmox');
+
+var client;
+
+if (process.env.S3_EMULATION) {
+  client = noxmox.mox.createClient({
+    key: 'local',
+    secret: 'host',
+    bucket: 'test'
+  });
+} else {
+  client = noxmox.nox.createClient({
+    key: process.env.AWS_ACCESS_KEY_ID,
+    secret: process.env.AWS_SECRET_ACCESS_KEY,
+    bucket: process.env.AWS_BUCKET
+  });
+}
 
 var log = require('../../logger.js');
+var File = require('../modules/Files/model');
+
+function modifyFiles(action, project) {
+  return File.query({
+    where: {
+      project_id: project.id
+    }
+  }).fetchAll()
+  .then(function(records){
+    var fn = action === 'publish' ? uploadFile : deleteFile;
 
-var isPublished = function(project) {
-  return project.get('publish_url') !== null;
+    return records.mapThen(function(record) {
+      return fn(record, project.get('user_id'));
+    });
+  })
+  .catch(function(e) {
+    return Promise.reject(e);
+  });
 };
 
+function uploadFile(file, userId) {
+  var deferred = new Promise();
```

Comment:
This is a convention I've seen in a few different places. Isn't a jquery `deferred` object a promise anyway?